src/rcc/enable.rs

```diff
@@ -25,6 +45,26 @@ macro_rules! bus_smenable {
             fn sleep_mode_disable(rcc: &mut Rcc) {
                 Self::Bus::smenr(rcc).modify(|_, w| w.$smen().clear_bit());
             }
+            #[inline(always)]
+            fn is_sleep_mode_enabled() -> bool {
+                let rcc = unsafe { &*RCC::ptr() };
+                Self::Bus::smenr(rcc).read().$smen().bit_is_set()
+            }
+            #[inline(always)]
+            fn is_sleep_mode_disabled() -> bool {
+                let rcc = unsafe { &*RCC::ptr() };
+                Self::Bus::smenr(rcc).read().$smen().bit_is_clear()
+            }
+            #[inline(always)]
+            unsafe fn sleep_mode_enable_unchecked() {
+                let rcc = &*RCC::ptr();
+                Self::Bus::smenr(rcc).modify(|_, w| w.$smen().set_bit());
+            }
+            #[inline(always)]
+            unsafe fn sleep_mode_disable_unchecked() {
+                let rcc = &*RCC::ptr();
+                Self::Bus::smenr(rcc).modify(|_, w| w.$smen().clear_bit());
+            }
```

Comment:
What is the main use case for those? I can see that they use `RCC` internally.

Reply:
The main reason of `unsafe` unchecked versions is to give ease to use API for places where `Rcc` is not available. Fortunately it is not used in this hal, but can be required by user.